pymeasure/instruments/agilent/agilent8722ES.py

```diff
@@ -121,48 +131,51 @@ def set_IF_bandwidth(self, bandwidth):
                                  "8722ES exceeded")
 
     def set_averaging(self, averages):
-        """ Turns on averaging of a specific number between 0 and 999
-        """
-        if int(averages) > 999 or int(averages) < 0:
-            raise RangeException("Averaging must be in the range 0 to 999")
+        """Sets the number of averages and enables/disables averaging"""
+        averages = int(averages)
+        if not 1 <= averages <= 999:
+            assert RangeException("Invalid number of averages -", averages)
+        self.averages = averages
+        if averages > 1:
+            self.enable_averaging()
         else:
-            self.write("AVERO1")
-            self.write("AVERFACT%d" % int(averages))
+            self.disable_averaging()
 
     def disable_averaging(self):
-        """ Disables averaging """
+        """Disables averaging"""
         self.write("AVERO0")
 
+    def enable_averaging(self):
+        """Enables averaging"""
+        self.write("AVERO1")
+
     def is_averaging(self):
         """ Returns True if averaging is enabled """
         return self.ask("AVERO?") == '1\n'
 
-    def restart_averaging(self, averages):
```

Comment:
I would suggest leaving the existing functions and properties (I did not mark the other occasions this remark would hold for) in existence (albeit as a wrapper for a new, updated function), with the same outcome / functionality as before, such that people using this instrument can keep relying on the functionality as they used to do.

Reply:
Agreed, we should not suddenly remove functionality. In methods, you can emit a warning about this with an appropriate message instructing the user how to avoid the warning: `warnings.warn(message, FutureWarning)`.

Reply:
I opted to refactor things a little more aggressively as it seems the module was already broken. There's some functions that were renamed back in April 2015 with commit 0c4597ba, and at least the scan function has been non-functional since then.

Should I go back and make sure things are as compatible as possible with the existing functions, or do you think it doesn't matter too much? I should get a chance to look at the other stuff soon.